### PR TITLE
Reduce context allocations

### PR DESCRIPTION
**Pull request by odersky**

### CODE REVIEW

**Comment by odersky** [comment]
test performance please

**Comment by odersky** [comment]
test performance please

**Comment by odersky** [comment]
test performance please.

**Comment by odersky** [comment]
check performance please

**Comment by odersky** [comment]
test performance please

File: compiler/src/dotty/tools/dotc/typer/Typer.scala [hunk 1]
```diff
@@ -3548,7 +3548,7 @@ class Typer extends Namer
   /** Types the body Scala 2 macro declaration `def f = macro <body>` */
   private def typedScala2MacroBody(call: untpd.Tree)(using Context): Tree =
     // TODO check that call is to a method with valid signature
-    def typedPrefix(tree: untpd.RefTree)(splice: Context ?=> Tree => Tree)(using Context): Tree = {
+    def typedPrefix(tree: untpd.RefTree)(splice: Ctx[Tree] => Tree)(using Context): Tree = {
```
Comment (nicolasstucki) [inline compiler/src/dotty/tools/dotc/typer/Typer.scala:3551]:
```suggestion
    def typedPrefix(tree: untpd.RefTree)(splice: Ctx[Tree => Tree])(using Context): Tree = {
```

**Comment by odersky** [comment]
Dropped in favor of #9312

**Pull request closed without merging**